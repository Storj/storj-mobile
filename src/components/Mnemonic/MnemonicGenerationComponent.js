import {
    View,
    Text,
    Image,
    TouchableOpacity,
    StyleSheet,
    Clipboard,
    Alert
} from 'react-native';
import React, { Component } from 'react';
import mnemonicScreenConstants from '../../utils/constants/mnemonicScreenConstants';
import { getWidth, getHeight } from '../../utils/adaptive';
import { getMnemonic } from '../../utils/AsyncStorageModule';

/**
* MnemonicGeneration component
*/
export default class MnemonicGenerationComponent extends Component {
    constructor(props) {
        super(props);

        /**
         * Local state for changing rendered view
         */
        this.state = {
            mnemonic: null,
            showCopyPopUp: false
        }

        this.secondWordsRowIndexCorrection = 13;
    }

    async componentDidMount() {

        getMnemonic().then((res)=>{
            this.setState({mnemonic: res});
        })
    }

    copyToClipboard() {
        Clipboard.setString( this.state.mnemonic );
        this.setState({showCopyPopUp: true});
    }

    cancelCopy() {
        this.setState({showCopyPopUp: false});
    }

    mnemonicToArrayView(mnemonic) {
        let array = mnemonic.split(' ');
        let firstArray = array.slice(0,12);
        let secondArray = array.slice(12);

        return (
            <View style = { styles.mnemonicContainer }>
                <View>
                    {
                        firstArray.map((element, index) => {
                            return (
                                <View key = { index } style = { styles.flexRow }>
                                    <Text style = { styles.index } >{ index + 1 }</Text>
                                    <Text style = { styles.word }>{ element }</Text>
                                </View>
                            )
                        })
                    }
                </View>
                <View style = { styles.secondArrayContainer }>
                    {
                        secondArray.map((element, index) => {  
                            return (
                                <View key = { index } style = { styles.flexRow }>
                                    <Text style = { styles.index } >{ index + this.secondWordsRowIndexCorrection }</Text>
                                    <Text style = { styles.word }>{ element }</Text>
                                </View>
                            )
                        })
                    }
                </View>
            </View>
        )
    }

    render() {
        return(
            <View style = { styles.mainContainer }>
                <View style = { styles.topContainer } >
                    <View style = { styles.topContentContainer } >
                        <View style = { styles.flexRow }>
                            <TouchableOpacity 
                                onPress = { () => { this.props.screenProps.redirectToMnemonicInfoScreen(); } }
                                style = { styles.backButtonContainer } >
                                <Image 
                                    source = { require('../../images/MyAccount/BlueBackButton.png') }
                                    style = { styles.icon } />
                            </TouchableOpacity>
                            <View>
                                <Text style = { [styles.titleText, styles.titleMargin] }>Secret phrase</Text>
                            </View>
                        </View>
                        <TouchableOpacity 
                            onPress = { () => { this.props.screenProps.redirectToLoginScreen(); } }
                            style = { styles.backButtonContainer } >
                            <Text style = { [styles.cancelText, styles.titleMargin] }>Skip</Text>
                        </TouchableOpacity>
                    </View>
                </View>
                <Text style = { styles.lesserTitleText }>Your secret phrase</Text>
                {
                    this.state.mnemonic 
                        ? this.mnemonicToArrayView(this.state.mnemonic)
                        : null
                }
                <TouchableOpacity onPress = { this.copyToClipboard.bind(this) }>
                    <View style = { styles.copyToClipboardContainer }>
                        <Image
                            style = { styles.icon }
                            source = { require('../../images/RegisterInfoScreens/Clipboard.png') }
                            resizeMode = 'contain' />
                        <Text style = { styles.clipboardText }>Copy to clipboard</Text>
                    </View>
                </TouchableOpacity>
                <TouchableOpacity onPress = { () => { this.props.screenProps.redirectToMnemonicConfirmationScreen(this.state.mnemonic); } } >
                    <View style = { styles.nextButton } >
                        <Text style = { styles.nextButtonText }>Next</Text>
                    </View>
                </TouchableOpacity>
                {
                    this.state.showCopyPopUp 
                    ? <View style = { styles.popUpContainer } >
                        <Text style = { styles.popUpInfoText } >Copied to clipboard</Text>
                        <TouchableOpacity onPress = { this.cancelCopy.bind(this) } >
                            <Text style = { styles.popUpCancelText }>OK</Text>
                        </TouchableOpacity>
                    </View> : null
                }
            </View>
        )
    }
}

const styles = StyleSheet.create({
    mainContainer: {
        flex: 1, 
        backgroundColor: '#FFFFFF',
        paddingHorizontal: getWidth(20)
    },
    topContainer: {
        height: getHeight(115)
    },
    topContentContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginTop: getHeight(15)
    },
    backButtonContainer: {
        justifyContent: 'flex-start',
        alignItems: 'center',
        marginTop: getHeight(6),
        height: getHeight(100)
    },
    flexRow: {
        flexDirection: 'row'
    },
    titleText: { 
        fontFamily: 'Montserrat-Bold', 
        fontSize: getHeight(30), 
        lineHeight: getHeight(33),
        color: '#384B65' 
    },
    titleMargin: {
        marginLeft: getWidth(20),
    },
    icon: {
        height: getHeight(24),
        width: getWidth(24)
    },
    cancelText: {
        fontFamily: 'Montserrat-Medium', 
        fontSize: getHeight(18), 
        lineHeight: getHeight(22),
        color: '#2794FF'
    },
    lesserTitleText: {
        fontFamily: 'Montserrat-Medium', 
        fontSize: getHeight(12), 
        lineHeight: getHeight(15),
        color: 'rgba(56, 75, 101, 0.4)'
    },
    mnemonicContainer: {
        flexDirection: 'row',
        marginTop: getHeight(15),
        height: getHeight(305)
    },
    secondArrayContainer: {
        marginLeft: getWidth(75)
    },
    index: {
        fontFamily: 'Montserrat-Regular', 
        fontSize: getHeight(16), 
        lineHeight: getHeight(25),
        color: 'rgba(56, 75, 101, 0.4)'
    },
    word: {
        marginLeft: getWidth(20),
        fontFamily: 'Montserrat-Regular', 
        fontSize: getHeight(16), 
        lineHeight: getHeight(25),
        color: '#384B65'
    },
    copyToClipboardContainer: {
        marginTop: getHeight(45),
        height: getHeight(55),
        flexDirection: 'row',
        alignItems: 'center'
    },
    clipboardText: {
        marginLeft: getWidth(15),
        fontFamily: 'Montserrat-Regular', 
        fontSize: getHeight(16), 
        lineHeight: getHeight(20),
        color: '#2794FF'
    },
    nextButton: {
        marginTop: getHeight(25),
        alignSelf: 'center',
        width: getWidth(335),
        height: getHeight(50),
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#2782ff',
        borderColor: '#2794FF',
        borderRadius: getWidth(6),
        borderWidth: getWidth(1.5)
    },
    nextButtonText: {
        fontFamily: 'Montserrat-Bold',
        fontSize: getHeight(14),
        color: 'white'
    },
    popUpContainer: {
        position: 'absolute',
        bottom: getHeight(10),
        alignSelf: 'center',
        width: getWidth(355),
        height: getHeight(55),
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        backgroundColor: '#FFFFFF',
        borderColor: '#FFFFFF',
        borderRadius: getWidth(6),
        borderWidth: getWidth(1.5),
        elevation: 5,
        paddingHorizontal: getWidth(20)
    },
    popUpInfoText: {
        fontFamily: 'Montserrat-Regular', 
        fontSize: getHeight(16), 
        lineHeight: getHeight(20),
        color: '#384B65',
        elevation: 5
    },
    popUpCancelText: {
        fontFamily: 'Montserrat-Regular', 
        fontSize: getHeight(16), 
        lineHeight: getHeight(20),
        color: '#2794FF'
    }
});