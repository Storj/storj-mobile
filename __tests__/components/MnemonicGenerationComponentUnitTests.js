import {
    View,
    TouchableOpacity,
    Text,
    TextInput,
    Alert
} from 'react-native';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'; 
import sinon from 'sinon';
import MnemonicGenerationComponent from '../../src/components/MnemonicGenerationComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('MnemonicGenerationComponent', () => {

    it('renders correctly', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with pressOut', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        const expectedState = { mnemonic: null,
            email: null,
            isLoading: true,
            showMnemonic: true,
            showModal: false,
            isMnemonicCopied: false,
            showConfirmation: false };

        wrapper.find('TouchableOpacity').forEach((child) =>{
            child.simulate('PressOut')
        });

        expect(wrapper.state()).toEqual(expectedState);
    });

    it('renders correctly with default state', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        expect(wrapper.containsMatchingElement(<Text>GET MNEMONIC</Text>)).toBe(true);
    });

    it('renders correctly with changing state to { isLoading: false, showMnemonic: true, mnemonic: someValue }', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        wrapper.setState({ isLoading: false, showMnemonic: true, mnemonic: 'someValue' });

        expect(wrapper.containsMatchingElement(<TextInput value = { wrapper.state().mnemonic } />)).toBe(true);
    });

    it('renders correctly with changing state to { showModal: true }', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        wrapper.setState({ showModal: true });

        expect(wrapper.containsMatchingElement(<Text>Copy to Clipboard</Text>)).toBe(true);
    });

    it('renders correctly with changing state to { showModal: false, isMnemonicCopied: true }', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        wrapper.setState({ showMnemonic : true, showModal: false, isMnemonicCopied: true, isLoading: false });

        expect(wrapper.containsMatchingElement(<Text>CONTINUE</Text>)).toBe(true);
    });

    it('renders correctly with changing state to { showConfirmation: true }', () => {

        const wrapper = shallow(
            <MnemonicGenerationComponent />
        );

        wrapper.setState({ showConfirmation: true });

        expect(wrapper.containsMatchingElement(Alert)).toBe(true);
    });
});