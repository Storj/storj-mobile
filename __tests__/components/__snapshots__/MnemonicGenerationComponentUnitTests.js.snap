// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicGenerationComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MnemonicGenerationComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          resizeMode="contain"
          source="getMnemonicLogo.png"
          style={
            Object {
              "height": 500,
              "left": -80,
              "position": "absolute",
              "top": 206,
              "width": 500,
            }
          }
        />,
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "position": "absolute",
              "right": 32,
              "top": 40,
              "width": 280,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat",
                "fontSize": 28,
                "lineHeight": 48,
                "marginBottom": 10,
              }
            }
          >
            What is Mnemonic?
          </Text>
          <Image
            source="Info.png"
            style={
              Object {
                "height": 48,
                "width": 48,
              }
            }
          />
        </View>,
        <View
          style={
            Object {
              "height": 84,
              "marginLeft": 32,
              "marginTop": 580,
              "width": 350,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat-ExtraBold",
                "fontSize": 92,
              }
            }
          >
            Get
          </Text>
        </View>,
        <View
          style={
            Object {
              "height": 84,
              "marginLeft": 32,
              "marginTop": 16,
              "width": 640,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat-ExtraBold",
                "fontSize": 92,
              }
            }
          >
            mnemonic
          </Text>
        </View>,
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginLeft": 32,
              "marginTop": 20,
              "width": 700,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "height": 328,
                "marginTop": 20,
                "width": 640,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#384b65",
                  "fontFamily": "Montserrat",
                  "fontSize": 32,
                }
              }
            >
              An explanation text tells your audience how something works or why something happens.
            </Text>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPressOut={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#2782ff",
                "borderRadius": 12,
                "height": 88,
                "justifyContent": "center",
                "marginTop": 48,
                "width": 686,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 28,
                }
              }
            >
              GET MNEMONIC
            </Text>
          </TouchableOpacity>
        </View>,
        null,
        null,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "resizeMode": "contain",
          "source": "getMnemonicLogo.png",
          "style": Object {
            "height": 500,
            "left": -80,
            "position": "absolute",
            "top": 206,
            "width": 500,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat",
                  "fontSize": 28,
                  "lineHeight": 48,
                  "marginBottom": 10,
                }
              }
            >
              What is Mnemonic?
            </Text>,
            <Image
              source="Info.png"
              style={
                Object {
                  "height": 48,
                  "width": 48,
                }
              }
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "space-between",
            "position": "absolute",
            "right": 32,
            "top": 40,
            "width": 280,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "What is Mnemonic?",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat",
                "fontSize": 28,
                "lineHeight": 48,
                "marginBottom": 10,
              },
            },
            "ref": null,
            "rendered": "What is Mnemonic?",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": "Info.png",
              "style": Object {
                "height": 48,
                "width": 48,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat-ExtraBold",
                "fontSize": 92,
              }
            }
          >
            Get
          </Text>,
          "style": Object {
            "height": 84,
            "marginLeft": 32,
            "marginTop": 580,
            "width": 350,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Get",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#2782ff",
              "fontFamily": "Montserrat-ExtraBold",
              "fontSize": 92,
            },
          },
          "ref": null,
          "rendered": "Get",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat-ExtraBold",
                "fontSize": 92,
              }
            }
          >
            mnemonic
          </Text>,
          "style": Object {
            "height": 84,
            "marginLeft": 32,
            "marginTop": 16,
            "width": 640,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "mnemonic",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#2782ff",
              "fontFamily": "Montserrat-ExtraBold",
              "fontSize": 92,
            },
          },
          "ref": null,
          "rendered": "mnemonic",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "height": 328,
                  "marginTop": 20,
                  "width": 640,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#384b65",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                  }
                }
              >
                An explanation text tells your audience how something works or why something happens.
              </Text>
            </View>,
            <TouchableOpacity
              activeOpacity={0.2}
              onPressOut={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderRadius": 12,
                  "height": 88,
                  "justifyContent": "center",
                  "marginTop": 48,
                  "width": 686,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                GET MNEMONIC
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 32,
            "marginTop": 20,
            "width": 700,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#384b65",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                  }
                }
              >
                An explanation text tells your audience how something works or why something happens.
              </Text>,
              "style": Object {
                "alignSelf": "flex-start",
                "height": 328,
                "marginTop": 20,
                "width": 640,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "An explanation text tells your audience how something works or why something happens.",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#384b65",
                  "fontFamily": "Montserrat",
                  "fontSize": 32,
                },
              },
              "ref": null,
              "rendered": "An explanation text tells your audience how something works or why something happens.",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "getMnemonic",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                GET MNEMONIC
              </Text>,
              "onPressOut": [Function],
              "style": Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#2782ff",
                "borderRadius": 12,
                "height": 88,
                "justifyContent": "center",
                "marginTop": 48,
                "width": 686,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "GET MNEMONIC",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 28,
                },
              },
              "ref": null,
              "rendered": "GET MNEMONIC",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            resizeMode="contain"
            source="getMnemonicLogo.png"
            style={
              Object {
                "height": 500,
                "left": -80,
                "position": "absolute",
                "top": 206,
                "width": 500,
              }
            }
          />,
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "space-between",
                "position": "absolute",
                "right": 32,
                "top": 40,
                "width": 280,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat",
                  "fontSize": 28,
                  "lineHeight": 48,
                  "marginBottom": 10,
                }
              }
            >
              What is Mnemonic?
            </Text>
            <Image
              source="Info.png"
              style={
                Object {
                  "height": 48,
                  "width": 48,
                }
              }
            />
          </View>,
          <View
            style={
              Object {
                "height": 84,
                "marginLeft": 32,
                "marginTop": 580,
                "width": 350,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat-ExtraBold",
                  "fontSize": 92,
                }
              }
            >
              Get
            </Text>
          </View>,
          <View
            style={
              Object {
                "height": 84,
                "marginLeft": 32,
                "marginTop": 16,
                "width": 640,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat-ExtraBold",
                  "fontSize": 92,
                }
              }
            >
              mnemonic
            </Text>
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginLeft": 32,
                "marginTop": 20,
                "width": 700,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "height": 328,
                  "marginTop": 20,
                  "width": 640,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#384b65",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                  }
                }
              >
                An explanation text tells your audience how something works or why something happens.
              </Text>
            </View>
            <TouchableOpacity
              activeOpacity={0.2}
              onPressOut={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderRadius": 12,
                  "height": 88,
                  "justifyContent": "center",
                  "marginTop": 48,
                  "width": 686,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  }
                }
              >
                GET MNEMONIC
              </Text>
            </TouchableOpacity>
          </View>,
          null,
          null,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": "getMnemonicLogo.png",
            "style": Object {
              "height": 500,
              "left": -80,
              "position": "absolute",
              "top": 206,
              "width": 500,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#2782ff",
                    "fontFamily": "Montserrat",
                    "fontSize": 28,
                    "lineHeight": 48,
                    "marginBottom": 10,
                  }
                }
              >
                What is Mnemonic?
              </Text>,
              <Image
                source="Info.png"
                style={
                  Object {
                    "height": 48,
                    "width": 48,
                  }
                }
              />,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "position": "absolute",
              "right": 32,
              "top": 40,
              "width": 280,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "What is Mnemonic?",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat",
                  "fontSize": 28,
                  "lineHeight": 48,
                  "marginBottom": 10,
                },
              },
              "ref": null,
              "rendered": "What is Mnemonic?",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": "Info.png",
                "style": Object {
                  "height": 48,
                  "width": 48,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat-ExtraBold",
                  "fontSize": 92,
                }
              }
            >
              Get
            </Text>,
            "style": Object {
              "height": 84,
              "marginLeft": 32,
              "marginTop": 580,
              "width": 350,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Get",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat-ExtraBold",
                "fontSize": 92,
              },
            },
            "ref": null,
            "rendered": "Get",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#2782ff",
                  "fontFamily": "Montserrat-ExtraBold",
                  "fontSize": 92,
                }
              }
            >
              mnemonic
            </Text>,
            "style": Object {
              "height": 84,
              "marginLeft": 32,
              "marginTop": 16,
              "width": 640,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "mnemonic",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#2782ff",
                "fontFamily": "Montserrat-ExtraBold",
                "fontSize": 92,
              },
            },
            "ref": null,
            "rendered": "mnemonic",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "height": 328,
                    "marginTop": 20,
                    "width": 640,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#384b65",
                      "fontFamily": "Montserrat",
                      "fontSize": 32,
                    }
                  }
                >
                  An explanation text tells your audience how something works or why something happens.
                </Text>
              </View>,
              <TouchableOpacity
                activeOpacity={0.2}
                onPressOut={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#2782ff",
                    "borderRadius": 12,
                    "height": 88,
                    "justifyContent": "center",
                    "marginTop": 48,
                    "width": 686,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 28,
                    }
                  }
                >
                  GET MNEMONIC
                </Text>
              </TouchableOpacity>,
            ],
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginLeft": 32,
              "marginTop": 20,
              "width": 700,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#384b65",
                      "fontFamily": "Montserrat",
                      "fontSize": 32,
                    }
                  }
                >
                  An explanation text tells your audience how something works or why something happens.
                </Text>,
                "style": Object {
                  "alignSelf": "flex-start",
                  "height": 328,
                  "marginTop": 20,
                  "width": 640,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "An explanation text tells your audience how something works or why something happens.",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#384b65",
                    "fontFamily": "Montserrat",
                    "fontSize": 32,
                  },
                },
                "ref": null,
                "rendered": "An explanation text tells your audience how something works or why something happens.",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "getMnemonic",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Montserrat-Bold",
                      "fontSize": 28,
                    }
                  }
                >
                  GET MNEMONIC
                </Text>,
                "onPressOut": [Function],
                "style": Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#2782ff",
                  "borderRadius": 12,
                  "height": 88,
                  "justifyContent": "center",
                  "marginTop": 48,
                  "width": 686,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "GET MNEMONIC",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "white",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 28,
                  },
                },
                "ref": null,
                "rendered": "GET MNEMONIC",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
